src/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -214,22 +214,13 @@ async Task<MsalAccessTokenCacheItem> ITokenCacheInternal.FindAccessTokenAsync(Au
 
             if (requestParams.AuthorityInfo != null)
             {
-                var instanceDiscoveryMetadataEntry = await GetCachedOrDiscoverAuthorityMetaDataAsync(
-                    requestParams.AuthorityInfo.CanonicalAuthority,
-                    requestParams.RequestContext).ConfigureAwait(false);
+                InstanceDiscoveryMetadataEntry instanceDiscoveryMetadataEntry =
+                    await ServiceBundle.InstanceDiscoveryManager.GetMetadataEntryAsync(
+                        new Uri(requestParams.AuthorityInfo.CanonicalAuthority),
+                        requestParams.RequestContext)
+                    .ConfigureAwait(false);
 
-                environmentAliases.UnionWith(GetEnvironmentAliases(
-                    requestParams.AuthorityInfo.CanonicalAuthority,
-                    instanceDiscoveryMetadataEntry));
-
-                if (requestParams.AuthorityInfo.AuthorityType == AuthorityType.Adfs)
```

Comment:
TokenCache is no longer responsible for knowing which authority can make instance discovery or not